src/lib/analyzeOas.ts

```diff
@@ -0,0 +1,213 @@
+import type { OASAnalysis, OASAnalysisFeature } from 'oas/dist/analyzer';
+import type { OASDocument } from 'oas/dist/rmoas.types';
+
+import analyzer from 'oas/dist/analyzer';
+
+export interface AnalyzedFeature extends OASAnalysisFeature {
+  description: string;
+  url?:
+    | string
+    | {
+        /**
+         * OpenAPI 3.1 introduced some new features so there won't be any docs on 3.0.
+         */
+        '3.0'?: string;
+        '3.1': string;
+      };
+}
+
+export interface Analysis extends OASAnalysis {
+  openapi: {
+    additionalProperties: AnalyzedFeature;
+    callbacks: AnalyzedFeature;
+    circularRefs: AnalyzedFeature;
+    discriminators: AnalyzedFeature;
+    links: AnalyzedFeature;
+    style: AnalyzedFeature;
+    polymorphism: AnalyzedFeature;
+    serverVariables: AnalyzedFeature;
+    webhooks: AnalyzedFeature;
+    xml: AnalyzedFeature;
+  };
+  readme: {
+    'x-default': AnalyzedFeature;
+    'x-readme.headers': AnalyzedFeature;
+    'x-readme.proxy-enabled': AnalyzedFeature;
+    'x-readme.samples-languages'?: AnalyzedFeature;
+    'x-readme.code-samples': AnalyzedFeature;
+    'x-readme.explorer-enabled': AnalyzedFeature;
+
+    /**
+     * This extension is deprecated.
+     */
+    'x-readme.samples-enabled'?: AnalyzedFeature;
+
+    /**
+     * RAW_BODY is specific to our Manual API editor and we don't recommend anyone writing their
+     * own API definition should use it so this is considered deprecated.
+     */
+    raw_body?: AnalyzedFeature;
+  };
```

Comment:
Reason I did it this way is because I have this type in `oas` because that's where all these queries exist and I'm augmenting it to be `AnalyzedFeature` because I didn't want to store the `description` and `url` information for each of these over in that repo and it didn't feel right living there.